### ISSUE TITLE
Convolutions and Pooling

### ISSUE THREAD

**Issue by MikeInnes** [JuliaGPU/GPUArrays.jl#102]
See [here](https://github.com/JuliaImages/ImageFiltering.jl/issues/52) for a description of the operations involved.

I expect that pooling will be relatively easy, convolutions are also straightforward enough in principle  but hard to make really fast. We probably want to port or wrap an existing high-quality implementation for Open CL, but I'm not sure what the benchmarks is here (perhaps [DeepCL](https://github.com/hughperkins/DeepCL)?).

**Comment by tejank10** [comment]
I am interested to take this up! From what I interpret, do we require the GPUArray version of the convolutions and pooling in Julia? Please correct me if I am wrong.

**Comment by MikeInnes** [comment]
Yes, that or some wrapper for an existing implementation. A generic kernel is probably a good idea either way though.

**Comment by XVilka** [comment]
@tejank10 have you had any luck with this?
